File: SWEA/D2/1986. 지그재그 숫자/지그재그 숫자.py
T = int(input())  # 테스트 케이스 수

for t in range(1, T + 1):
    N = int(input())  # 입력

    result = 0  # 누적 합

    # 1부터 N까지 반복
    for i in range(1, N + 1):
        if i % 2 == 0:
            result -= i  # 짝수는 빼기
        else:
            result += i  # 홀수는 더하기

    print(f"#{t} {result}")  # 출력

File: SWEA/D2/1986. 지그재그 숫자/지그재그 숫자.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt(); // 테스트 케이스 수

        for (int t = 1; t <= T; t++) {
            int N = sc.nextInt(); // 숫자 N 입력

            int result = 0; // 누적 결과 저장 변수

            // 1부터 N까지 반복
            for (int i = 1; i <= N; i++) {
                if (i % 2 == 0) {
                    result -= i; // 짝수면 빼기
                } else {
                    result += i; // 홀수면 더하기
                }
            }

            // 출력 형식
            System.out.println("#" + t + " " + result);
        }
    }
}